WapTenantPublicAPI.psm1

```diff
@@ -2715,6 +2870,18 @@ function New-WAPVM {
                 )
             }
 
+            if ($HWProfile) {
+                $Body.add("HardwareProfileId", $HWProfile.id)
+            }
+
+            if ($CPU) {
+                $Body.add("CPUCount", $CPU)
+            }
+
+            if ($Memory) {
```

Comment:
when memory is not specified by the user, it will have the value 0. You should use ```PSBoundParameters.ContainsKey('Memory')```